### PR TITLE
use fd 0 for communication between components

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
Here's my version of #14060

### CODE REVIEW

File: doc/man/cockpit-bridge.xml [hunk 1]
```diff
@@ -82,6 +82,14 @@
             available to the user running this command.</para>
         </listitem>
       </varlistentry>
+      <varlistentry>
```
Comment (martinpitt) [inline doc/man/cockpit-bridge.xml:85]: The Closes # still refers to my original PR. Please adjust.

File: src/ws/session-utils.c [hunk 2]
```diff
@@ -225,7 +225,7 @@ write_control_end (void)
   assert (auth_msg_size > 0);
   assert (auth_msg != NULL);
 
-  if (cockpit_frame_write (STDOUT_FILENO, (unsigned char *)auth_msg, auth_msg_size) < 0)
+  if (cockpit_frame_write (STDIN_FILENO, (unsigned char *)auth_msg, auth_msg_size) < 0)
```
Comment (martinpitt) [inline src/ws/session-utils.c:228]: This looks confusing now. Perhaps this should define a new constant `BRIDGE_FD`?

File: src/ws/session-utils.h [hunk 3]
```diff
@@ -62,10 +62,6 @@ void build_string (char **buf, size_t *size, const char *str, size_t len);
 void authorize_logger (const char *data);
 void save_environment (void);
 void pass_to_child (int signo);
-#ifndef HAVE_FDWALK
-int fdwalk (int (*cb)(void *data, int fd), void *data);
-#endif
-int closefd (void *data, int fd);
```
Comment (martinpitt) [inline src/ws/session-utils.h:68]: IMO we should keep the closing of all fds > 3. This is a security boundary, so let's not take chances from leaking fds to kerberos or other PAM services.

Comment (allisonkarlitskaya) [reply src/ws/session-utils.h]: Brought this back.

File: doc/man/cockpit-bridge.xml (hunk 1 above)
Comment (allisonkarlitskaya) [reply doc/man/cockpit-bridge.xml:85]: Done.

File: src/ws/session-utils.c (hunk 2 above)
Comment (allisonkarlitskaya) [reply src/ws/session-utils.c]: added to `cockpitframe.h`

File: doc/man/cockpit-bridge.xml (hunk 1 above)
Comment (martinpitt) [reply doc/man/cockpit-bridge.xml:85]: No, it's still wrong. :-)

File: src/ssh/cockpitsshrelay.c [hunk 4]
```diff
@@ -2242,9 +2234,12 @@ cockpit_ssh_relay_start (CockpitSshRelay *self,
   in = dup (0);
   g_assert (in >= 0);
 
+  out = dup (0);
+  g_assert (out >= 0);
```
Comment (martinpitt) [inline src/ssh/cockpitsshrelay.c:2238]: Why is that necessary? Can't in-fd and out-fd both use the same fd? (and `in` would then be renamed to `inout` or maybe `bridge_fd`)

Comment (allisonkarlitskaya) [reply src/ssh/cockpitsshrelay.c:2238]: CockpitPipe takes ownership and wants to close each of them independently.

**Comment by allisonkarlitskaya** [comment]
There's an extremely suspicious

```
dd: failed to open '/dev/stdin': No such device or address
```

Leaking out of `test-auth` `/auth/idle-timeout`

**Comment by allisonkarlitskaya** [comment]
hm.  The unit tests worked when I ran them locally, but it's pretty clear that absolutely none of the builders are happy with them, and it seems to come down to this same `dd` problem.  I'll take a look after lunch.

**Comment by allisonkarlitskaya** [comment]
Current state, in summary:

`tcsh` closes all fds above stdout/in/err.

`tlog-rec-session` remaps stdin/out/err through a pty.

running `tcsh` under `tlog-rec-session` will leave no fds untouched.

sshd seems to open stdin/out/err as pipes.

**Comment by allisonkarlitskaya** [comment]
This is probably replaced by #14375.

**Comment by allisonkarlitskaya** [comment]
I just merged the other PR.
